rcl/src/rcl/wait.c

```diff
@@ -79,6 +80,9 @@ __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
     assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
   }
   if (wait_set->impl) {
+    if (wait_set->impl->rmw_waitset) {
+      assert(rmw_destroy_waitset(wait_set->impl->rmw_waitset) == RMW_RET_OK);
```

Comment:
Since the existing implementations of `rmw_destroy_waitset` do return an error code in certain situations I think the return value should definitely be checked here (meaning also in release). Otherwise the rmw impl. might detect a problem but it stays unnoticed here.